### ISSUE TITLE
develop: WhichKey not found

### ISSUE THREAD

**Issue by bketelsen** [NTBBloodbath/doom-nvim#26]
I followed the instructions to run the develop branch, and neovim reports:

`E492: Not an editor command: WhichKey \`

I tried with a fresh clone, removed .local/share/nvim..., did it manually and with the install script, both have the same error message.

**Comment by NTBBloodbath** [comment]
Hey, a friend (doom-nvim contributor) has also had some problems with other plugins in the development branch but I have been unable to get them even with a fresh install :confused:. 

I'll create an Arch VM to test there. I hope there neovim will send me the errors to be able to give debugging. I will leave this issue open until then. Thanks for letting me know about this issue.

**Comment by bketelsen** [comment]
Thanks! I've tried on macos/arm, windows/WSL and Ubuntu 21.04 with the same results.

**Comment by NTBBloodbath** [comment]
Hey, I was finally able to debug on my machine (without the VM, VirtualBox was screwing up my Arch ISO)!

I have did a few tweaks that I hope can fix that issue:

1. Fixed a plugin lazy-loading, that gets rid of some startup issues when the plugins are installed.
2. Packer bootstrapping is now managed directly by doom's source instead of the installer, that fixes an inconsistency.
3. Dashboard will not be called on first launch because obviously the plugin is not installed at this point so there will be no more errors about it.
4. The automatic installation of plugins on launch is a really good feature, but was having some issues when triggered on first launch, should be fixed now. See why about it below.

> Explanation of change 4:

Basically this feature triggers `PackerInstall` on launch and that's ok. It installs the missing plugins so the end user does not have to trigger it manually so we can automatize it and spend more time editing.

But `PackerInstall` does not run `PackerCompile` nor compile the packer plugins so the plugins are not being loaded until the user compiles the plugins with `PackerCompile` and that's not ok. That's why we are using `PackerSync` now on first launch.

With `PackerSync` we improve a lot that feature because it will install the missing plugins, update the existing ones (this one does not apply on first launch, let's ignore this point) and then run `PackerCompile`. That's a great deal, but since it updates the plugins we are only using it on first launch so doom-nvim will not try to update your plugins if you don't want to, and keep the feature as it should be, for just installing missing plugins.

---

> **NOTE**: If you are going to install the fix using the installer, use the develop branch as it contains the changes from change 2. Just change the `main` to `develop` in the URL of the curl command. I have also added your changes to it so the installer should work fine.

There are a lot of things fixed on first launch but I have not faced that WhichKey error, let me know if these also fixes that WhichKey issue.

Regards
